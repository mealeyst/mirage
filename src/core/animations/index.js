export { default as FadeInOut } from './FadeInOut'
