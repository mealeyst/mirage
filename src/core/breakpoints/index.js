import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Desktop = styled(({element, children, ...props}) => {
  return React.createElement(element, props, children)
})`
display: none;
${props => props.theme.breakpointsVerbose.aboveTabletMax`
  display: ${props => props.display};
`}
`

Desktop.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
Desktop.defaultProps = {
  display: 'block',
  element: 'div'
}

const Laptop = styled(({element, children, ...props}) => {
  return React.createElement(element, props, children)
})`
display: none;
${props => props.theme.breakpointsVerbose.aboveLaptop`
  display: ${props => props.display};
`}
`

Laptop.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
Laptop.defaultProps = {
  display: 'block',
  element: 'div'
}

const Tablet = styled(({element, children, ...props}) => {
  delete props.laptopDisplay
  return React.createElement(element, props, children)
})`
display: none;
${props => props.theme.breakpointsVerbose.aboveTablet`
  display: ${props => props.display};
`}
${props => props.theme.breakpointsVerbose.aboveLaptop`
  display: ${props => props.laptopDisplay};
`}
`

Tablet.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
Tablet.defaultProps = {
  display: 'block',
  element: 'div',
  laptopDisplay: 'none'
}

const TabletMax = styled(({element, children, ...props}) => {
  delete props.laptopDisplay
  return React.createElement(element, props, children)
})`
  display: none;
  ${props => props.theme.breakpointsVerbose.aboveTabletMax`
    display: ${props => props.display};
  `}
  ${props => props.theme.breakpointsVerbose.aboveLaptop`
    display: ${props => props.laptopDisplay};
  `}
`

TabletMax.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
TabletMax.defaultProps = {
  display: 'block',
  element: 'div',
  laptopDisplay: 'none'
}

const Mobile = styled(({element, children, ...props}) => {
  return React.createElement(element, props, children)
})`
display: ${props => props.display};
${props => props.theme.breakpointsVerbose.aboveTablet`
  display: none;
`}
`

Mobile.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node
}
Mobile.defaultProps = {
  display: 'block',
  element: 'div'
}

const Default = styled(({element, children, ...props}) => {
  delete props.displayTarget
  return React.createElement(element, props, children)
})`
display: none;
${props => props.theme.breakpointsVerbose[props.displayTarget]`
    display: ${props => props.display};
`}
`

Default.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  element: PropTypes.node,
  displayTarget: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
}
Default.defaultProps = {
  display: 'block',
  element: 'div',
  displayTarget: 'belowTabletMax'
}

export {
  Default,
  Desktop,
  Laptop,
  Mobile,
  Tablet,
  TabletMax
}
