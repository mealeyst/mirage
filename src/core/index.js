export * from './grid'
export * from './icons'
export * from './image'
export * from './typography'
export * from './theme'
export * from './video'
