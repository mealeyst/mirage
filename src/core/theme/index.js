import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

export { theme }

export default class ThemeWrapper extends Component {
  render () {
    return (
      <ThemeProvider theme={theme} {...this.props}>
        {this.props.children}
      </ThemeProvider>
    )
  }
}

ThemeWrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.string
  ]).isRequired
}
