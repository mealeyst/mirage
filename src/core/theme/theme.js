import { injectGlobal } from 'styled-components'
import { colors } from './colors'
import media, { breakpoints, breakpointsVerbose } from './mediaQueries'
import fontFamilies from './fontFamilies'
import gridSettings from './gridSettings'

const base = injectGlobal`
  html {
    font-size: 10px;
  }
`

export default {
  breakpoints: breakpoints,
  colors: colors,
  fonts: fontFamilies,
  grid: gridSettings,
  fixedPosition: 'relative',
  media: media,
  breakpointsVerbose: breakpointsVerbose,
  base: base
}
