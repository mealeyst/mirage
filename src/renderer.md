```js
const data = require('./pages/home.json');
<Renderer item={data} />
```
