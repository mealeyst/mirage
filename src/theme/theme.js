import { colors } from './colors'
import fontFamilies from './fontFamilies'

export default {
  colors: colors,
  fonts: fontFamilies,
  fixedPosition: 'relative'
}
