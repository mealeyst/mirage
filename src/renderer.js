import React from 'react'
import PropTypes from 'prop-types'

import * as Core from './core'
import * as Components from './components'
import * as Modules from './modules'


const COMPONENT_RENDER_MAP = {
  Core: {
    BackgroundImage: (item) => {
      return (
        <Core.BackgroundImage {...item.props}>
          {item.children.map(child => Renderer({item: child}))}
        </Core.BackgroundImage>
      )
    },
    Caption: (item) =>
      <Core.Caption {...item.props}>
        {item.children}
      </Core.Caption>,
    H1: (item) =>
      <Core.H1 {...item.props}>
        {item.children}
      </Core.H1>,
    H2: (item) =>
      <Core.H2 {...item.props}>
        {item.children}
      </Core.H2>,
    H3: (item) =>
      <Core.H3 {...item.props}>
        {item.children}
      </Core.H3>,
    H4: (item) =>
      <Core.H4 {...item.props}>
        {item.children}
      </Core.H4>,
    H5: (item) =>
      <Core.H5 {...item.props}>
        {item.children}
      </Core.H5>,
    H6: (item) =>
      <Core.H6 {...item.props}>
        {item.children}
      </Core.H6>,
    Label: (item) =>
      <Core.Label {...item.props}>
        {item.children}
      </Core.Label>,
    P: (item) =>
      <Core.P {...item.props}>
        {item.children}
      </Core.P>,
    Grid: (item) =>
      <Core.Grid>
        {item.children.map(child => Renderer({item: child}))}
      </Core.Grid>,
    Logo: () => <Core.Logo />,
    Sizer: (item) =>
      <Core.Sizer {...item.props}>
        {item.children.map(child => Renderer({item: child}))}
      </Core.Sizer>
  },
  Components: {
    InformationalSection: (item) =>
      <Components.InformationalSection>
        {item.children.map(child => Renderer({item: child}))}
      </Components.InformationalSection>,
    BackgroundVideo: (item) =>
      <Components.BackgroundVideo {...item.props}>
        {item.children.map(child => Renderer({item: child}))}
      </Components.BackgroundVideo>
  },
  Modules: {
    Page: (item) =>
      <Modules.Page>
        {item.children.map(child => Renderer({item: child}))}
      </Modules.Page>
  }
}

const Renderer = ({item}) => {
  try {
    return COMPONENT_RENDER_MAP[item.sys.type][item.sys.component](item.data)
  } catch(err) {
    console.warn(`It appears that you are tying to render an element that doesn't exist in the COMPONENT_RENDER_MAP`)
  }
}

Renderer.propTypes = {
  item: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object
  ])
}

export default Renderer
