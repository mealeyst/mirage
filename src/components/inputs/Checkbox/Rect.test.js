import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { css, keyframes } from 'styled-components'
import 'jest-styled-components'

import { theme } from 'SRC/core/theme'
import StyledRect, { checked, unchecked } from './Rect.base'

describe('(Base Component) Rect', () => {
  test('matching the snapshot', () => {
    const component = shallow(<StyledRect theme={theme} />)
    expect(component).toMatchSnapshot()
  })

  test('adding "checked" styles when checked is passed', () => {
    const component = shallow(<StyledRect theme={theme} checked />)
    expect(component).toHaveStyleRule({
      modifier: css`${checked}`,
    })
  })

  test('adding "unchecked" styles when checked is not passed', () => {
    const component = shallow(<StyledRect theme={theme} />)
    expect(component).toHaveStyleRule({
      modifier: css`${unchecked}`,
    })
  })
})
