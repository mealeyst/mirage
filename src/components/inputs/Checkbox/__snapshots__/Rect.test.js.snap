// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Base Component) Rect matching the snapshot 1`] = `
.c0 {
  -webkit-animation: hBmzvO 0.25s linear forwards;
  animation: hBmzvO 0.25s linear forwards;
  stroke: transparent;
  stroke-width: 10;
  stroke-linecap: round;
  stroke-dashoffset: 200;
  fill: none;
  stroke: #0073D1;
  stroke-width: 20;
  stroke-linecap: round;
}

<Rect
  className="c0"
  theme={
    Object {
      "base": undefined,
      "breakpoints": Object {
        "aboveDesktop": "(min-device-width: 1440px)",
        "aboveLaptop": "(min-device-width: 1280px)",
        "abovePhone": "(min-device-width: 320px)",
        "abovePhoneMax": "(min-device-width: 414px)",
        "aboveTablet": "(min-device-width: 768px)",
        "aboveTabletMax": "(min-device-width: 960px)",
        "belowDesktop": "(max-device-width: 1440px)",
        "belowLaptop": "(max-device-width: 1280px)",
        "belowPhone": "(max-device-width: 320px)",
        "belowPhoneMax": "(max-device-width: 414px)",
        "belowTablet": "(max-device-width: 768px)",
        "belowTabletMax": "(max-device-width: 960px)",
        "landscape": "(orientation: landscape)",
        "portrait": "(orientation: portrait)",
      },
      "colors": Object {
        "black": "#000000",
        "electricBlue": "#080CF1",
        "gray": Array [
          "#F3F3F3",
          "#F5F5F5",
          "#F8F8F8",
          "#E6E6E6",
          "#D5D5D5",
          "#CFCFCF",
        ],
        "gray0": "#F3F3F3",
        "gray1": "#F5F5F5",
        "gray2": "#F8F8F8",
        "gray3": "#E6E6E6",
        "gray4": "#D5D5D5",
        "gray5": "#CFCFCF",
        "lightBlue": Array [
          "#EEFCFF",
          "#BCE1F7",
          "#76C2F3",
        ],
        "lightBlue0": "#EEFCFF",
        "lightBlue1": "#BCE1F7",
        "lightBlue2": "#76C2F3",
        "lightPink": "#FBDDE4",
        "loading": "#BFBFBF",
        "navy": "#00003C",
        "pink": "#FEACBE",
        "red": "#FF511C",
        "rocketBlue": "#0073D1",
        "rocketBlueHover": "#0073EB",
        "white": "#FFFFFF",
        "yellow": "#F0FF02",
        "yellowHover": "#D8E320",
      },
      "fixedPosition": "relative",
      "fonts": Object {
        "headerFont": "\\"din-cond\\", \\"din-condensed-web\\", Arial, sans-serif",
        "primaryFont": "\\"ff-din-web\\", Helvetica, Arial, sans-serif",
        "secondaryFont": "\\"adobe-caslon-pro\\", Helvetica, Arial, serif",
      },
      "grid": Object {
        "columns": Object {
          "desktop": 12,
          "mobile": 4,
          "tablet": 10,
        },
        "gutter": "10px",
        "margins": Object {
          "desktop": "7%",
          "mobile": "3%",
          "tablet": "7%",
        },
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "phoneMax": [Function],
        "tablet": [Function],
        "tabletMax": [Function],
      },
    }
  }
/>
`;
